### PR TITLE
Integrate LLVM at llvm/llvm-project@23ed570af1cc

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Update BUILD files and submodule for
[23ed570af1cc](https://github.com/llvm/llvm-project/commit/23ed570af1cc). Includes updates necessary for
[7a996027b984](https://github.com/llvm/llvm-project/commit/7a996027b984)

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
This includes updates necessary for two separate commits that happened to be near each other so I manually stepped through history. I'm still trying to provide functioning build files for arbitrary commits, so would like to merge this with a rebase.

File: llvm-bazel/llvm-project-overlay/llvm/BUILD [hunk 1]
```diff
@@ -194,6 +194,7 @@ cc_library(
     }),
     hdrs = glob([
         "include/llvm/Support/*.h",
+        "include/llvm/Support/FileSystem/*.h",
```
Comment (GMNGeoffrey) [inline llvm-bazel/llvm-project-overlay/llvm/BUILD]: Another option would be just globbing for `"include/llvm/Support/**/*.h"` which I think wouldn't be too unreasonable.

Comment (chandlerc) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: I'm good with either.

**Review by chandlerc — APPROVED** [review]
LGTM

File: llvm-bazel/llvm-project-overlay/llvm/BUILD (hunk 1 above)
Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: Decided to go with the globbier glob as likely to be more future-proof

**Pull request merged**